test_cases/stdlib/builtins/test_list.py

```diff
@@ -0,0 +1,16 @@
+from typing_extensions import assert_type
+
+
+# list.__add__ example from #8292
+class Foo:
+    def asd(self) -> int:
+        return 1
+
+
+class Bar:
+    def asd(self) -> int:
+        return 2
+
+
+combined = [Foo()] + [Bar()]
+assert_type(combined, list[Foo | Bar])
```

Comment:
We have to use old syntax in the test suite, since it's a `.py` file and we test on 3.7+

```suggestion
assert_type(combined, List[Union[Foo, Bar]])
```